From be5eeaddcbbb479568f3cf64809669062455031a Mon Sep 17 00:00:00 2001
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Mon, 7 Jan 2013 16:50:11 -0500
Subject: [PATCH] wrhv: arm: mach-vexpress support CPU hotplug

Change points:
    - paravirt the cpu die routine
    - create a paravirt routine to weakly use the native one

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 arch/arm/mach-vexpress/hotplug.c |   10 +++++++++-
 1 files changed, 9 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-vexpress/hotplug.c b/arch/arm/mach-vexpress/hotplug.c
index c9d5036..d427951 100644
--- a/arch/arm/mach-vexpress/hotplug.c
+++ b/arch/arm/mach-vexpress/hotplug.c
@@ -98,7 +98,7 @@ int platform_cpu_kill(unsigned int cpu)
  *
  * Called with IRQs disabled
  */
-void platform_cpu_die(unsigned int cpu)
+void native_platform_cpu_die(unsigned int cpu)
 {
 	int spurious = 0;
 
@@ -118,6 +118,14 @@ void platform_cpu_die(unsigned int cpu)
 		pr_warning("CPU%u: %u spurious wakeup calls\n", cpu, spurious);
 }
 
+void paravirt_platform_cpu_die(unsigned int cpu)
+	__attribute__((weak, alias("native_platform_cpu_die")));
+
+void platform_cpu_die(unsigned int cpu)
+{
+	paravirt_platform_cpu_die(cpu);
+}
+
 int mach_cpu_disable(unsigned int cpu)
 {
 	/*
-- 
1.7.0

